src/go/rpk/pkg/adminapi/api_transform.go

```diff
@@ -69,3 +69,17 @@ func (a *AdminAPI) ListWasmTransforms(ctx context.Context) ([]TransformMetadata,
 	err := a.sendAny(ctx, http.MethodGet, baseTransformEndpoint, nil, &resp)
 	return resp, err
 }
+
+type patchMetadataRequest struct {
+	IsPaused    *bool                  `json:"is_paused,omitempty"`
+	Environment *[]EnvironmentVariable `json:"env,omitempty"`
+}
+
+// PatchTransformMetadata sets transform paused state and/or overwrites transform environment vars
+func (a *AdminAPI) PatchTransformMetadata(ctx context.Context, transformName string, paused *bool, env *[]EnvironmentVariable) error {
```

Comment:
Why do we need a pointer to a bool? Could it be `bool` and that way we avoid the function that you use in the command to pass a pointer?

Reply:
We could today, but it would cost some generality. At some point (soon) we'll want the ability to live-update a transform's environment variables, possibly without altering its "paused" state. So by passing a pointer we maintain the "optionality" of each param at the interface.

An alternative would be to have wrapper functions specifically for Pause (and later PatchEnv or whatever) that delegate to the more general version, like

```go
func (a *AdminAPI) SetTransformPaused(ctx context.Context, transformName string, paused bool) error {
    return PatchTransformMetadata(ctx, transformName, &paused, nil)
}
```

Wdyt?

Reply:
Oh, in that case, wdyt of having `PauseTransform` and `ResumeTransform`, similar to what we do with Roles:

https://github.com/redpanda-data/redpanda/blob/dev/src/go/rpk/pkg/adminapi/api_security.go#L103-L119

Reply:
Ah, yeah that seems fine. Will do. 

Editorial: I wish golang had `optional`s 😇